### PR TITLE
Fix - TGM Plugin activation calling undefined function wc_get_screen_ids

### PR DESCRIPTION
**Pull request by shivapoudel**
If we add woocommerce and enable the `force_activation` config in TGM plugin activation then the bug appears.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

image

### How to test the changes in this Pull Request:

All this does is ensure that a particular file is loaded before a function within that file is called. Most of the time, the file is already loaded, so in those cases, this change does nothing. Rather than having to install and set up the third-party TGM Plugin, all we really need to do here is make sure this change doesn't break anything. So after switching to this branch, try a few things:

* Deactivating and reactivating the plugin
* Navigate to a few different WooCommerce pages within WP Admin

Doing these things, you shouldn't see any fatal errors about notices or about functions already being defined.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Ensure WC admin functions are defined before attempting to add notices.

#### Comment 

</details>

### CODE REVIEW

**Review by coreymckrill — COMMENTED** [review]
Hi @shivapoudel, thanks for the PR! Rather than bailing early on the `add_notices` method when `wc_get_screen_ids` doesn't exist, I think it would be preferable to maintain the functionality and just ensure that `wc_get_screen_ids` _does_ exist.

The reason the error is happening is that the `wc-admin-functions.php` file isn't getting loaded during the TGM request for some reason. But we could do something like what happens [here](https://github.com/woocommerce/woocommerce/blob/83fc7a88f35e3382c17a244ce24d349fa9d6d08f/plugins/woocommerce/includes/tracks/events/class-wc-products-tracking.php#L13-L13), and require the file within `add_notices`. So maybe at the top of the method we could add a line like this:

```php
require_once WC_ABSPATH . 'includes/admin/wc-admin-functions.php';
```

What do you think?

**Comment by shivapoudel** [comment]
But TGM is external library so how the inclusion occurs? Or you mean to say rather than `function_exists` check just to include the file within `wc_get_screen_ids` function.

File: plugins/woocommerce/includes/admin/class-wc-admin-notices.php [hunk 1]
```diff
@@ -301,7 +301,7 @@ public static function user_has_dismissed_notice( string $name, ?int $user_id =
 	public static function add_notices() {
 		$notices = self::get_notices();
 
-		if ( empty( $notices ) ) {
+		if ( empty( $notices ) || ! function_exists( 'wc_get_screen_ids' ) ) {
 			return;
 		}
```
Comment (coreymckrill) [inline plugins/woocommerce/includes/admin/class-wc-admin-notices.php]:
```suggestion
		if ( empty( $notices ) ) {
			return;
		}

		require_once WC_ABSPATH . 'includes/admin/wc-admin-functions.php';
```

**Review by coreymckrill — COMMENTED** [review]
> Or you mean to say rather than function_exists check just to include the file within wc_get_screen_ids function.

Sort of. I mean something like the below suggestion.

File: plugins/woocommerce/includes/admin/class-wc-admin-notices.php (hunk 1 above)
Comment (coreymckrill) [reply plugins/woocommerce/includes/admin/class-wc-admin-notices.php]: Does this fix the TGM issue?

Comment (shivapoudel) [reply plugins/woocommerce/includes/admin/class-wc-admin-notices.php]: Yes. This fixes the issue.

**Comment by shivapoudel** [comment]
@coreymckrill Adapted the changes. Thanks!

**Comment by coreymckrill** [comment]
Closing and re-opening this PR to fix the stuck CI actions...

**Review by coreymckrill — APPROVED** [review]

**Pull request merged**